R/usage_to_annual_nets_distributed.R

```diff
@@ -0,0 +1,234 @@
+                                  data.frame(iso3=NA, month=NA, access=1, 
+                                             percapita_nets = 1, loess_predicted_access=1))
+    print("Access-nets per capita curve is extrapolated linearly beyond observed levels")
+  } else {
+    print("No extrapolation beyond observed access-nets per capita relationship - return NA")
+  }
+  
+  ### 1. Convert annual nets distributed per capita to equilibrium nets per capita ###
+  
+  # Create dataframe of all combinations of input countries and nets distributed
+  nets_distributed <- expand.grid(iso3 = country_iso3,
+                                  annual_percapita_nets_distributed = annual_percapita_nets_distributed)
+  
+  # Net half lives (median retention times) for each country
+  half_life_data <- data$half_life_data
+  
+  # Estimate time at which all nets are lost (parameter l) based on net half life in each country
+  half_life_data$l <- half_life_data$half_life/sqrt(1 - k / (k - log(0.5)))
+  
+  # Merge this with annual net distribution by country 
+  nets_distributed <- merge(x=nets_distributed, y=half_life_data[,c("iso3", "l")], 
+                            by = "iso3", all.x=TRUE, all.y=FALSE)
+  
+  # Use the net loss function to calculate th equilibrium nets per capita for the given
+  # annual nets distribution per capita by country
+  # This is for the specific "smoothed" case where we assume that 1/3 of the population at risk
+  # is distributed nets every 3 years (this is equivalent to a smoothed 3 yearly whole 
+  # population distribution).
+  nets_distributed$target_percapita_nets <- nets_distributed$annual_percapita_nets_distributed/ 
+    (integrate_net_loss_vector(k=k, l = nets_distributed$l) * (1/3))
+  
+  ### 2. Convert nets per capita to target usage ###
+  
+  # Use Loess curve to convert nets per capita to access
+  nets_distributed$target_access <- 
+    approx(x=loess_for_prediction$percapita_nets, 
+           y=loess_for_prediction$loess_predicted_access,
+           nets_distributed$target_percapita_nets)$y
+  
+  # Convert target access to target usage: usage = access * use rate
+  # since use rate = the proportion of people with access to a net who slept under it
+  # Relationship between access and NPC is aggregated for all countries,
+  # but use rate to convert usage to access is country-specific (derived from spatiotemporal regression?)
+  nets_distributed <- merge(x=nets_distributed, y=country_use_rate, 
+                            by = "iso3", all.x=TRUE, all.y=FALSE)
+  nets_distributed$target_usage <- nets_distributed$target_access*nets_distributed$use_rate
+  
+  nets_distributed <- nets_distributed[order(nets_distributed$iso3, nets_distributed$annual_percapita_nets_distributed),]
+  
+  return(nets_distributed[,c("iso3", "use_rate", "annual_percapita_nets_distributed",
+                             "target_percapita_nets","target_access", 
+                             "target_usage")])
+  
+  # TO DO:
```

Comment:
Good to keep comments like this out of the code - they have a tendancy to sneek through to propoer releases. Feel free to stick stuff like this in the issues + conmversations on those issues.

Extrpolating at the lower end would be helpful yes.
I think we can have percapita nets >1 (its a bit crazy, but not impossible)